lib/ansible/modules/windows/win_optional_feature.ps1

```diff
@@ -0,0 +1,108 @@
+#!powershell
+
+# Copyright: (c) 2019, Carson Anderson <rcanderson23@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#AnsibleRequires -CSharpUtil Ansible.Basic
+
+$spec = @{
+	options = @{
+		name = @{ type = "str"; required = $true }
+		state = @{ type = "str"; default = "present"; choices = @("absent", "present") }
+		source = @{ type = "str" }
+		include_parent = @{ type = "bool"; default = $false }
+	}
+	supports_check_mode = $true
+}
+
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+$module.Result.rc = 0
+
+$name = $module.Params.name
+$state = $module.Params.state
+$source = $module.Params.source
+$include_parent = $module.Params.include_parent
+
+$win_version = (Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\").ProductName
+if ($win_version -notlike "Windows 10*") {
+    $module.FailJson("This version of Windows does not support the Enable-WindowsOptionalFeature.")
+}
+
+$feature_state_start = Get-WindowsOptionalFeature -Online -FeatureName $name
+if (-not $feature_state_start) {
+    $module.Result.rc = 1
+    $module.FailJson("Failed to find feature.")
+}
+
+
+if ($state -eq "present") {
+    if ($feature_state_start.State -notlike "Enabled*") {
```

Comment:
Any reason why you have the `Enabled*` wildcard, the value should be an enum?

Reply:
I wasn't sure how to handle this regarding "Enabled" vs "EnabledPending". I treated them as if they were the same but this may be incorrect thinking.

Reply:
Hmm maybe a comment saying this should match `Enabled` or `EnabledPending`, either that or just do `$feature_state_start.State -notin @("Enabled", "EnabledPending")`

Reply:
I have added comments for both Enabled and Disabled for clarity on matches